### PR TITLE
Fixed merge of the original event and repetition

### LINKED ISSUE #524: Project Overview / Open clicked repetition instead of the first one
For now, when you clicked on the event, the first one will be opened. We need to open a particular repetition.

image

### LINKED ISSUE #543: Fix Event tab switching
now merging works wrong

### PR DESCRIPTION
**Pull request by GeekaN2**
The repetition was merged incorrectly, I fixed the customizer for the fields. 

Resolves #543
Resolves #524

### CODE REVIEW

File: cypress/integration/utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,221 @@
+import { HawkEventPayload, HawkEventRepetition } from '@/types/events';
```
Comment (neSpecc) [inline cypress/integration/utils.test.ts:1]: why this file called utils? it contains tests for event merging

File: cypress/integration/utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  user: {
+    id: 0,
+    name: '',
+    url: '',
+    photo: '',
+  },
+  addons: {}
+}
+
+type DeepNullability<T> = {
```
Comment (neSpecc) [inline cypress/integration/utils.test.ts:30]: describe in jsdoc please

File: cypress/integration/utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+  let originalEvent = baseEvent;
+  let repetitionEvent: RepetitionEventPayload = baseEvent;
+  let assembledRepetition: HawkEventPayload = baseEvent;
+
+  beforeEach(() => {
+    originalEvent = cloneDeep(baseEvent);
+    repetitionEvent = cloneDeep(baseEvent);
+  });
+
+  it('Should get object params from the original event', () => {
```
Comment (neSpecc) [inline cypress/integration/utils.test.ts:46]:
```suggestion
  it('should get object params from the original event', () => {
```

Comment (neSpecc) [reply cypress/integration/utils.test.ts:46]: > object params

what does it mean?

File: cypress/integration/utils.test.ts [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+      ...baseEvent,
+      release: 'Omega release',
+    };
+
+    const result = repetitionAssembler(originalEvent, repetitionEvent);
+
+    expect(result).to.deep.equal(assembledRepetition);
+  });
+
+  it('Should take the value from the repetition payload', () => {
```
Comment (neSpecc) [inline cypress/integration/utils.test.ts:67]: > Merging of the repetition and the original event payloads at the overview stage should take the value from the repetition payload

can't understand the spec

File: src/store/modules/events/index.ts [hunk 5]
```diff
@@ -497,14 +497,14 @@ const module: Module<EventsModuleState, RootState> = {
...
-        event,
+        event: {
+          ...event,
+          payload: event.repetition ? repetitionAssembler(event.payload, event.repetition.payload) as HawkEventPayload : event.payload,
+        },
       });
 
       const repetition = event.repetition;
 
-      filterBeautifiedAddons([ event ]);
```
Comment (neSpecc) [inline src/store/modules/events/index.ts:505]: why?

**Review by talyguryn — APPROVED** [review]

**Review by neSpecc — APPROVED** [review]
👍

**Pull request merged**